lib/app/dispatcher.rb

```diff
@@ -7,31 +7,32 @@
 # frozen_string_literal: true
 
 require 'active_support'
-require 'concurrent-ruby'
+require 'concurrent'
 require 'connectors'
 require 'core'
 require 'utility'
 require 'app/config'
 
 module App
   class Dispatcher
-    POLL_IDLING = (App::Config[:idle_timeout] || 60).to_i
-    TERMINATION_TIMEOUT = (App::Config[:termination_timeout] || 60).to_i
-    MIN_THREADS = (App::Config.thread_pool&.min_threads || 0).to_i
-    MAX_THREADS = (App::Config.thread_pool&.max_threads || 5).to_i
-    MAX_QUEUE = (App::Config.thread_pool&.max_queue || 100).to_i
+    POLL_INTERVAL = (App::Config.poll_interval || 3).to_i
+    TERMINATION_TIMEOUT = (App::Config.termination_timeout || 60).to_i
+    HEARTBEAT_INTERVAL = (App::Config.heartbeat_interval || 60 * 30).to_i
+    MIN_THREADS = (App::Config.dig(:thread_pool, :min_threads) || 0).to_i
+    MAX_THREADS = (App::Config.dig(:thread_pool, :max_threads) || 5).to_i
+    MAX_QUEUE = (App::Config.dig(:thread_pool, :max_queue) || 100).to_i
```

Comment:
What do you think about moving these lines to become private methods, just to show this way that they should not be used outside of `App::Dispatcher`? When they are a constant they give a feeling, that they can be used by anyone who import this class (which is not the case as far as I see)

Reply:
👍